### PR TITLE
KOGITO-840: [DMN Designer] Move support for client-side marshalling tests to new root module

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/KOGITO-840

This PR simply moves test support from an _integration_ module (invoked with the `maven-invoker` plugin) to a root module in `kie-wb-common-dmn` to overcome an issue with using the invoker plugin with parallel builds.

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-marshaller-tests/pom.xml [hunk 1]
```diff
@@ -25,8 +25,8 @@
   <modelVersion>4.0.0</modelVersion>
 
   <artifactId>kie-wb-common-dmn-webapp-kogito-marshaller-tests</artifactId>
-  <name>Kie Workbench - Common - DMN - Kogito Marshaller Integration Tests</name>
-  <description>Kie Workbench - Common - DMN - Kogito Marshaller Integration Tests</description>
+  <name>Kie Workbench - Common - DMN - Kogito Marshaller - Tests</name>
+  <description>Kie Workbench - Common - DMN - Kogito Marshaller - Tests</description>
   <packaging>jar</packaging>
 
   <dependencies>
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-marshaller-tests/pom.xml]:
```
[WARNING] Unused declared dependencies found:
[WARNING]    org.slf4j:slf4j-api:jar:1.7.26:compile
[WARNING]    org.slf4j:slf4j-ext:jar:1.7.26:compile
[WARNING]    ch.qos.logback:logback-classic:jar:1.2.3:compile
[WARNING]    org.slf4j:log4j-over-slf4j:jar:1.7.26:compile
[WARNING]    org.kie.workbench:kie-wb-common-dmn-webapp-common:jar:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench:kie-wb-common-dmn-webapp-kogito-common:jar:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-wires-core-grids:jar:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.soup:kie-soup-commons:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-client-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-project-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-structure-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-commons:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.uberfire:uberfire-nio2-model:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.stunner:kie-wb-common-stunner-core-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.stunner:kie-wb-common-stunner-core-common:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.stunner:kie-wb-common-stunner-client-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.stunner:kie-wb-common-stunner-client-common:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.stunner:kie-wb-common-stunner-kogito-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.stunner:kie-wb-common-stunner-kogito-client:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench:kie-wb-common-dmn-api:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench:kie-wb-common-dmn-client:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench:kie-wb-common-dmn-webapp-kogito-common:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench:kie-wb-common-dmn-webapp-kogito-marshaller:jar:sources:7.32.0-SNAPSHOT:compile
[WARNING]    org.kie.workbench.forms:kie-wb-common-forms-adf-base:jar:sources:7.32.0-SNAPSHOT:compile
```

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-marshaller-tests/pom.xml]: They are actually used.. I did not originally copy them from the `kie-wb-common-dmn-kogito-webapp-common` modules' `pom.xml` and encountered a lot of compilation errors generating the `GWTTestCase`.. I slowly added them back one by one to end until all the compilation errors (relating to missing _sources_).

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-marshaller-tests/pom.xml]: I'll look at the enforcer issue now....

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Pull request merged**